src/components/ImportRecord/ImportRecord.js

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage, injectIntl } from 'react-intl';
+import ky from 'ky';
+import { stripesConnect, withOkapiKy, CalloutContext } from '@folio/stripes/core';
+import { LoadingView } from '@folio/stripes/components';
+
+
+class ImportRecord extends React.Component {
+  static propTypes = {
+    id: PropTypes.string,
+    stripes: PropTypes.shape({
+      logger: PropTypes.object.isRequired,
+    }).isRequired,
+    resources: PropTypes.shape({
+      query: PropTypes.shape({
+        xid: PropTypes.string,
+      }).isRequired,
+    }).isRequired,
+    mutator: PropTypes.shape({
+      query: PropTypes.shape({
+        update: PropTypes.func.isRequired,
+      }).isRequired,
+    }).isRequired,
+    okapiKy: PropTypes.func.isRequired,
+    intl: PropTypes.shape({
+      formatMessage: PropTypes.func.isRequired,
+    }).isRequired,
+  };
+
+  static contextType = CalloutContext;
+
+  static manifest = Object.freeze({
+    query: {},
+  });
+
+  constructor(props) {
+    super(props);
+    this.loading = false;
+  }
+
+  componentDidUpdate() {
+    if (!this.loading) {
+      const xid = this.props.resources.query?.xid;
+      this.loading = true;
+      this.loadExternalRecord(xid);
+    }
+  }
+
+  loadExternalRecord = (xid) => {
+    const { id, intl } = this.props;
+
+    this.props.okapiKy('copycat/imports', {
+      timeout: 30000,
+      method: 'POST',
+      json: {
+        externalIdentifier: xid,
+        internalIdentifier: id,
+        profileId: 'f26df83c-aa25-40b6-876e-96852c3d4fd4' // XXX hardwired to UUID in mod-copycat reference data
+      },
+    }).then(res => {
+      // No need to check res.ok, as ky throws non-2xx responses
+      res.json().then(json => {
+        this.props.mutator.query.update({
```

Comment:
Please use: 
https://github.com/folio-org/ui-inventory/blob/master/src/withLocation.js

instead of query resource for navigation. That HOC should give you enough tools to do this easily. I think `goTo` should work fine in this case or if you want to preserve current params `updateLocation` should do it too.

After that I think the query mutator can be removed from here and also which means we should be able to remove `stripesConnect`.

Reply:
Resolved in https://github.com/folio-org/ui-inventory/commit/99ba2d27ea57207cdc4f52dbce7a9ce1873c6bca — the HOC did not give me the tools I needed, but I extended it to do so.